### PR TITLE
Parse a constraints string in prepareConstraints

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by makyo** [comment]
# QA

* make run
* Set a breakpoint at app/store/env/go.js:1022
* drag and drop http://pastebin.com/ULBjUtiy onto the canvas
* Ensure that request.Params.Constraints.tags exists and is an array

File: Makefile [hunk 1]
```diff
@@ -75,13 +75,13 @@ $(PY):
 $(NODE_MODULES):
 	npm install --cache-min 999999
 
-.PHONY: run-server
-run-server: gui
+.PHONY: server
```
Comment (makyo) [inline Makefile]: Drive-by per discussion

File: Makefile [hunk 2]
```diff
@@ -75,13 +75,13 @@ $(PY):
...
+server: gui
 	bin/pserve --reload development.ini
 
 .PHONY: run
-run:
-	make -j 2 run-server watch
+run: gui
+	$(MAKE) -j2 server watch
```
Comment (makyo) [inline Makefile]: Use this make process to start two additional threads.

**Comment by frankban** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2864/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA OK!

**Comment by makyo** [comment]
TY :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge
